modules/caddyhttp/matchers.go

```diff
@@ -477,18 +647,10 @@ func (MatchPathRE) CaddyModule() caddy.ModuleInfo {
 func (m MatchPathRE) Match(r *http.Request) bool {
 	repl := r.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
 
-	// PathUnescape returns an error if the escapes aren't
-	// well-formed, meaning the count % matches the RFC.
-	// Return early if the escape is improper.
-	unescapedPath, err := url.PathUnescape(r.URL.Path)
-	if err != nil {
-		return false
-	}
-
 	// Clean the path, merges doubled slashes, etc.
 	// This ensures maliciously crafted requests can't bypass
 	// the path matcher. See #4407
-	cleanedPath := path.Clean(unescapedPath)
+	cleanedPath := path.Clean(r.URL.Path)
```

Comment:
I think we can merge L653~L658 into one line:

```go
cleanedPath := cleanPath(r.URL.Path)
```

Reply:
Good point, thanks!

I do realize the path_regexp matcher works a little differently than (or rather, lacks some of the new flexibility of) the path matcher (as the path_regexp matcher always operates in the normalized space only), but that's probably OK...